modules/battery_manager/README.md

```diff
@@ -1,5 +1,76 @@
+   instrument is NEVER administered.
+   - If the test battery entry has a `stage` set and the `stage` matches the 
+   one of the timepoint, the instrument is administered at that stage.
+
+ - Center:
+   - If the test battery entry does NOT have a `CenterID` (`CenterID=NULL`), 
+   the instrument gets administered at ALL centers.
+   - If the test battery entry has a `CenterID` set and the `CenterID` matches
+   the one of the timepoint, the instrument is administered at that CenterID.
+
+### LEVEL medium:
+ - AgeMinDays/AgeMaxDays:
+   - If the test battery entry does NOT have a `AgeMinDays` or a `AgeMaxDays` 
+   or both (`AgeMinDays=NULL` or `AgeMaxDays=NULL`), the instrument is NEVER 
+   administered.
+   - If the test battery entry has `AgeMinDays` and `AgeMaxDays` set and they 
+   are both set to `0`, the instrument gets administered at ALL ages;
+   - If the test battery entry has `AgeMinDays` and `AgeMaxDays` set and they 
+   are set to any value other than `0`, the instrument gets administered IF AND
+   ONLY IF the age of the candidate at the timepoint is between `AgeMinDays` 
+   and `AgeMaxDays`;
+
+### LEVEL hard:
+
+ - VisitLabel:
+   - If the test battery entry does NOT have a `Visit_label`
+   (`Visit_label=NULL`), the instrument gets administered IF AND ONLY IF no
+   other test battery entries matches the timepoint's visit label and
+   subproject.
+   - If the test battery entry has a `Visit_label` set and the `Visit_label`
+   matches the one of the timepoint, the instrument is administered at that
+   Visit.
+   - ***NOTE:** In order to administer an instrument at all visits without
+   defining each visit individually in the battery, the test battery table
+   should NOT contain any entries for the subproject/visit_label combination
+   of the timepoint. If the timepoint's subproject/visit_label combination
+   has a specified set of instruments defined in the test_battery, all entries
+   of the battery with no visit labels (`Visit_label=NULL`) will be ignored.*
+ - FirstVisit:
+   - If the test battery entry does NOT have a `firstVisit` (`firstVisit=NULL`),
+   the instrument is administered based on the rules described above.
+   - If the test battery entry has a `firstVisit` set to YES (`Y`) and the
+   timepoint is the first session for the given candidate, the instrument is
+   administered regardless of any of the other rules described above.
+   - If the test battery entry has a `firstVisit` set to NO (`N`) and the
+   timepoint is the first session for the given candidate, the instrument is
+   NOT administered regardless of any of the other rules described above.
+   - If the test battery entry has a `firstVisit` set to YES (`Y`) and the
+   timepoint is NOT the first session for the given candidate, the instrument
+   is NOT administered regardless of any of the other rules described above.
+   - If the test battery entry has a `firstVisit` set to NO (`N`) and the
```

Comment:
I think all these first visit descriptions can be simplified. The intended behaviour is (in addition to the rules for the other columns):

1. If firstVisit is set to 'Y', the test battery instance is applied only if it is the first visit.
2. If firstVisit is set to 'N', the test battery instance is applied only if it is *not* the first visit.
3. If firstVisit is set to null, the test battery instance is applied to any visit.

Reply:
well its still important to mention
that if you have first visit to Y and you add a centerID in the row, that will do absolutely nothing

same for visitlabel age ranges stages...

Reply:
nvm the note is still there. I'm okai with that